cmd/mdatagen/loader.go

```diff
@@ -45,15 +50,13 @@ type metric struct {
 
 	// ExtendedDocumentation of the metric. If specified, this will
 	// be appended to the description used in generated documentation.
-	ExtendedDocumentation string `yaml:"extended_documentation"`
+	ExtendedDocumentation string `mapstructure:"extended_documentation"`
 
 	// Unit of the metric.
-	Unit string `yaml:"unit"`
+	Unit string
 
-	// Raw data that is used to set Data interface below.
-	YmlData *ymlMetricData `yaml:"data" validate:"required"`
 	// Date is set to generic metric data interface after validating.
-	Data MetricData `yaml:"-"`
+	Data MetricData `validate:"required"`
```

Comment:
I think we should do a bit different here maybe and adopt something similar with receivers protocols, see https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/jaegerreceiver/config.go#L44

Idea user will write:

```yaml
metrics:
  system.cpu.time:
    description: Total CPU seconds broken down by different states.
    unit: s
    sum:
      monotonic: true
      aggregation: cumulative
    attributes: [freeFormAttribute, freeFormAttributeWithValue, enumAttribute]
``

Reply:
I was also thinking about this interface. The only concern is that it allows having several datatypes since it's set as a key now, so we need to put some more validation in place

Reply:
We will have a "validation", much simpler than current logic I think, also much simpler for users to read what properties are accepted for specific types.

Reply:
Ok. I applied this approach in another PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6463. I agree, it looks better now. I'll update this one after